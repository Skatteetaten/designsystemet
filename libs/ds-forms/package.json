{
  "name": "@skatteetaten/ds-forms",
  "groupId": "no.skatteetaten.aurora",
  "version": "2.0.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^24",
    "react": "^19 ",
    "react-i18next": "^15",
    "@skatteetaten/ds-core-utils": "2.0.0",
    "@skatteetaten/ds-icons": "2.0.0",
    "@skatteetaten/ds-buttons": "2.0.0",
    "@skatteetaten/ds-typography": "2.0.0",
    "@skatteetaten/ds-progress": "2.0.0",
    "@skatteetaten/ds-status": "2.0.0",
    "date-fns": "^4",
    "@skatteetaten/ds-content": "2.0.0"
  },
  "scripts": {}
}
