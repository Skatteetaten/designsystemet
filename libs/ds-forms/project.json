{
  "name": "ds-forms",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-forms/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-forms",
        "tsConfig": "libs/ds-forms/tsconfig.lib.json",
        "project": "libs/ds-forms/package.json",
        "entryFile": "libs/ds-forms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-forms/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-forms/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-forms"],
      "options": {
        "jestConfig": "libs/ds-forms/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-forms/**/*.css",
          "libs/ds-forms/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-forms"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-forms {args.switches}"
      }
    }
  }
}
