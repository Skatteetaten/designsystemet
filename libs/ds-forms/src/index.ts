export * from './Checkbox/Checkbox';
export * from './Checkbox/Checkbox.types';
export * from './CheckboxGroup/CheckboxGroup';
export * from './DatePicker/DatePicker.types';
export * from './DatePicker/DatePicker';
export * from './Select/Select';
export * from './Select/Select.types';
export * from './SelectOption/SelectOption.types';
export * from './CheckboxGroup/CheckboxGroup.types';
export * from './ErrorMessage/ErrorMessage';
export * from './ErrorMessage/ErrorMessage.types';
export * from './ErrorSummary/ErrorSummary';
export * from './ErrorSummary/ErrorSummary.types';
export * from './Fieldset/Fieldset';
export * from './Fieldset/Fieldset.types';
export * from './FileUploader/FileUploader';
export * from './FileUploader/FileUploader.types';
export * from './Radio/Radio';
export * from './Radio/Radio.types';
export * from './RadioGroup/RadioGroup';
export * from './RadioGroup/RadioGroup.types';
export * from './TextArea/TextArea.types';
export * from './TextArea/TextArea';
export * from './TextField/TextField.types';
export * from './TextField/TextField';
export * from './SearchField/SearchField';
export * from './SearchField/SearchField.types';
