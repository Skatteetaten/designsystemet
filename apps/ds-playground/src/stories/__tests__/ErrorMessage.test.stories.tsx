import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { ErrorMessage } from '@skatteetaten/ds-forms';

const meta = {
  component: ErrorMessage,
  title: 'Tester/ErrorMessage',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
    },
    showError: {
      table: { disable: true },
    },
  },
} satisfies Meta<typeof ErrorMessage>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultText = 'Feilmelding';

const defaultArgs = {
  showError: false,
  children: defaultText,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5, B1)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    showError: true,
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({
    canvasElement,
  }: {
    canvasElement: HTMLElement;
  }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    const errorMessage = canvas.getAllByRole('generic')[2];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(errorMessage).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (A1, A2)',
  args: {
    ...defaultArgs,
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toBeInTheDocument();
    await expect(container).toHaveAttribute('aria-live', 'assertive');
    await expect(container).toHaveAttribute('aria-atomic');

    const textNode = container.querySelector('p');
    await expect(textNode).not.toBeInTheDocument();
  },
} satisfies Story;

export const WithPElement = {
  name: 'With <p>-element (B1)',
  args: {
    ...defaultArgs,
    showError: true,
  },
  argTypes: {
    showError: {
      table: { disable: false },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const textNode = canvas.getByText(defaultText);
    await expect(textNode.tagName).toBe('P');
  },
} satisfies Story;

const TemplateWithInput: StoryFn<typeof ErrorMessage> = (args) => (
  <div className={'errorMessageContainer'}>
    <ErrorMessage {...args} />
  </div>
);

export const WithLongText = {
  render: TemplateWithInput,
  name: 'With Long Text (A3)',
  args: {
    ...defaultArgs,
    showError: true,
    children:
      'Dette blir en veldig lang feilmelding for å teste om du oppfører seg om den skal.',
  },
  argTypes: {
    children: {
      table: { disable: false },
    },
  },
} satisfies Story;

export const WithLongTextAndBreaking = {
  render: TemplateWithInput,
  name: 'With Long Text And Breaking',
  args: {
    ...defaultArgs,
    showError: true,
    children:
      'Detteblirenveldiglangfeilmeldingforåtesteomduoppførersegomdenskal.',
  },
  argTypes: {
    children: {
      table: { disable: false },
    },
  },
} satisfies Story;
