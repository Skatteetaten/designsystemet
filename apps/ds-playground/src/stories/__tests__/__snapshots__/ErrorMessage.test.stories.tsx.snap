// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ErrorMessage Defaults play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
  >
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       lang="nb"
       data-testid="123ID"
       aria-live="assertive"
       aria-atomic="true"
  >
    <div class="_errorMessage_1kdf4_15 dummyClassname">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
        </path>
      </svg>
      <p class="_errorMessageText_1kdf4_30">
        Feilmelding
      </p>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithLongText smoke-test 1`] = `
<div data-test-block="true">
  <div class="errorMessageContainer">
    <div aria-live="assertive"
         aria-atomic="true"
    >
      <div class="_errorMessage_1kdf4_15 ">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
        <p class="_errorMessageText_1kdf4_30">
          Dette blir en veldig lang feilmelding for å teste om du oppfører seg om den skal.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithLongTextAndBreaking smoke-test 1`] = `
<div data-test-block="true">
  <div class="errorMessageContainer">
    <div aria-live="assertive"
         aria-atomic="true"
    >
      <div class="_errorMessage_1kdf4_15 ">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
        <p class="_errorMessageText_1kdf4_30">
          Detteblirenveldiglangfeilmeldingforåtesteomduoppførersegomdenskal.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithPElement play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
  >
    <div class="_errorMessage_1kdf4_15 ">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
        </path>
      </svg>
      <p class="_errorMessageText_1kdf4_30">
        Feilmelding
      </p>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithRef play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;
